--- /mnt/_build/Server/modules/ffmpeg/consumer/ffmpeg_consumer.cpp	2017-05-12 13:34:34.102580815 +0000
+++ ffmpeg_consumer_fix.cpp	2017-05-12 13:35:25.861685489 +0000
@@ -723,10 +723,12 @@
 			set_pixel_format(filt_vsink, requested_fmt);
 		}
 
+		/* fixes REMOVE <> crash ??
 		if (in_video_format_.width < 1280)
 			video_graph_->scale_sws_opts = "out_color_matrix=bt601";
 		else
 			video_graph_->scale_sws_opts = "out_color_matrix=bt709";
+		*/
 
 		configure_filtergraph(
 				*video_graph_,

